### PR TITLE
check if user is logged in before checking their subscription status

### PR DESCRIPTION
**Pull request by aintgoin2goa**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -47,39 +48,47 @@ app.get('/__opt-in', function(req, res){
...
-				},
-				referrer: referrerUrlObj
-			}
-		});
-	} else {
-		debug('Not setting cookie');
+	if(!flag || flag.isSwitchedOff){
+		debug('Not setting cookie, flag is off');
 		debug('- Flag:', flag);
-		debug('- user.isSubscriber():', user.isSubscriber());
```
Comment (commuterjoy) [inline app.js]: I think we should collect Metrics around each of these conditions.

Comment (aintgoin2goa) [inline app.js]: how do I do that?  using the beacon?

Can I deploy this fix first and then look at that?

Comment (commuterjoy) [inline app.js]: Sure

Comment (commuterjoy) [inline app.js]: `metrics.count('barrier.is-not-logged-in');`

`metrics.count('barrier.is-not-subscriber');`

`metrics.count('barrier.flag-switched-off');`

`metrics.count('barrier.ok');`

**Comment by aintgoin2goa** [comment]
Added metrics as 
`optin.failure.flag_off`  
`optin.failure.not_logged_in`   
`optin.failure.not_subscriber`  
`opt-in.success`

**Pull request merged**